cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/AbstractProgramController.java

```diff
@@ -416,94 +416,104 @@ private ListenerCaller(Listener listener, Executor executor) {
       this.executor = executor;
     }
 
+    private void execute(Runnable runnable, String methodName) {
+      try {
+        executor.execute(runnable);
+      } catch (Throwable t) {
```

Comment:
Sure. Will move it to `MultiListenerCaller`. However we will still need to handle it in the `addListener` method which directly makes `caller.init` method call bypassing `MultiListenerCaller`, right?